### PR TITLE
Meeting and brand name subjects

### PR DESCRIPTION
**Pull request by warrd**
### Issues

https://github.com/wellcometrust/platform/issues/3633
https://github.com/wellcometrust/platform/issues/3634

### Description

Add "meeting" (as `Meeting`) and "brand name" (as `Concept`) subjects from marc fields 611 / 652.

### CODE REVIEW

File: pipeline/transformer/transformer_sierra/src/main/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/subjects/SierraMeetingSubjects.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+// Use MARC field "611".
+//
+// *  Populate the platform "label" with the concatenated values of
+//    subfields a, c, and d.
+//
+// *  Populate "concepts" with a single value:
+//
+//    -   Create "label" from subfields a, c and d
+//    -   Set "type" to "Meeting"
+//    -   Use subfield 0 to populate "identifiers", if present.  Note the
```
Comment (jamesgorrie) [inline pipeline/transformer/transformer_sierra/src/main/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/subjects/SierraMeetingSubjects.scala]: This comment doesn't look like it's been implemented below?

File: pipeline/transformer/transformer_sierra/src/main/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/subjects/SierraSubjectsTransformer.scala [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    *
+    */
+  def createLabel(varField: VarField, subfieldTags: List[String]): String =
+    varField.subfields
+      .filter { subfield =>
+        subfieldTags.contains(subfield.tag)
+      }
+      .sortBy { _.tag }
+      .map { _.content }
+      .mkString(" ")
```
Comment (jamesgorrie) [inline pipeline/transformer/transformer_sierra/src/main/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/subjects/SierraSubjectsTransformer.scala]: I thought we weren't meant to be adding any values? Does the marc fields not already contain ` `?

File: pipeline/transformer/transformer_sierra/src/test/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/subjects/SierraMeetingSubjectsTest.scala [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      ),
+      Unidentifiable(
+        Subject(
+          label = "Second",
+          concepts = List(Unidentifiable(Meeting(label = "Second")))
+        )
+      )
+    )
+  }
+}
```
Comment (jamesgorrie) [inline pipeline/transformer/transformer_sierra/src/test/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/subjects/SierraMeetingSubjectsTest.scala]: We could wrap most these tests up into 1 table driven test.
If we're worried about the logging, we should probably try to fix the logging with the table test libs we use.

File: pipeline/transformer/transformer_sierra/src/main/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/subjects/SierraSubjectsTransformer.scala (hunk 2 above)
Comment (warrd) [reply pipeline/transformer/transformer_sierra/src/main/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/subjects/SierraSubjectsTransformer.scala]: Not sure I understand. This method already existed [here](https://github.com/wellcometrust/catalogue/blob/master/pipeline/transformer/transformer_sierra/src/main/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/subjects/SierraOrganisationSubjects.scala#L76-L77), I have just moved it up to the base trait.

It concatenates the content of various subfields into a single label (which we seem to do all over the place in the transformer)

File: pipeline/transformer/transformer_sierra/src/test/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/subjects/SierraMeetingSubjectsTest.scala (hunk 3 above)
Comment (warrd) [reply pipeline/transformer/transformer_sierra/src/test/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/subjects/SierraMeetingSubjectsTest.scala]: Agree that these tests maybe a bit verbose, but still not convinced that table driven tests are the way to go. Each of the tests are actually testing different things, so even if the structure of the tests is similar I reckon makes sense to have separate unit tests.

**Pull request merged**